network/src/web.ts

```diff
@@ -0,0 +1,77 @@
+import { WebPlugin, PluginListenerHandle } from '@capacitor/core';
+import { NetworkPlugin, NetworkStatus } from './definitions';
+
+declare var window: any;
+
+export class NetworkWeb extends WebPlugin implements NetworkPlugin {
+  listenerFunction: any = null;
+
+  constructor() {
+    super({ name: 'Network' });
+  }
+
+  getStatus(): Promise<NetworkStatus> {
+    return new Promise((resolve, reject) => {
+      if (!window.navigator) {
+        reject('Network info not available');
+        return;
+      }
+
+      let connected = window.navigator.onLine;
+      let connection =
+        window.navigator.connection ||
+        window.navigator.mozConnection ||
+        window.navigator.webkitConnection;
+      let connectionType = connection
+        ? connection.type || connection.effectiveType
+        : 'wifi';
+
+      resolve({
+        connected: connected,
+        connectionType: connected ? connectionType : 'none',
+      } as NetworkStatus);
```

Comment:
Using `x as X` is a forceful cast. To avoid it, assign object literals to a variable with an explicit type.

```suggestion
      const status: NetworkStatus = {
        connected: connected,
        connectionType: connected ? connectionType : 'none',
      }

      resolve(status);
```